src/nvim/api/ui_events.in.h

```diff
@@ -35,6 +35,10 @@ void set_title(String title)
   FUNC_API_SINCE(3);
 void set_icon(String icon)
   FUNC_API_SINCE(3);
+void set_cursor(void)
+  FUNC_API_SINCE(3);
+void reset_cursor(void)
+  FUNC_API_SINCE(3);
```

Comment:
`mode_info_set` already specifies the cursor styles, and presumably the UI can react to `mode_change`. Why do we need new UI events?